### PR TITLE
Revert "Do not materialize full text/image/speech logits (10GB for 16k seq text)"

### PR DESCRIPTION
**Pull request by markblee**
This reverts commit 2e3c00fdbb9af83b5bd1721d2bd3dcb37b5db6ea.

### CODE REVIEW

**Review by ruomingp — APPROVED** [review]

**Review by ds-hwang — APPROVED** [review]
Thank you for quick action.
